src/zos_files/zowe/zos_files_for_zowe_sdk/uss.py

```diff
@@ -175,9 +189,20 @@ def get_content_streamed(self, file_path: str, binary: bool = False) -> dict:
         if binary:
             custom_args["headers"]["X-IBM-Data-Type"] = "binary"
         response = self.request_handler.perform_request("GET", custom_args, stream=True)
-        return response
+        if binary:
+            return {
+                "content_type": response.headers.get("Content-Type"),
+                "content_length": len(response.content),
+                "file_content": response.content.decode("utf-8", errors="ignore")  # You can also base64-encode this if needed
```

Comment:
This seems like it could break output for binary files because even though the content is binary, we try to decode the response content as UTF-8.

Reply:
This seems to be addressed now 🥳